python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -63,9 +63,16 @@ def __eq__(self, other):
         return self.path == other.path
 
     def to_string(self) -> str:
-        """E.g. '["first_component", "second_component"]'."""
+        """E.g. '["first_component", "second_component"]'.
+
+        Note: This method is used for serializing AssetKeys across several database tables,
+        and should not be changed.
+        """
         return seven.json.dumps(self.path)
 
+    def to_db_string(self) -> str:
+        return self.to_string()
```

Comment:
Yeah I put the "scary sounding" comments in there / the test because we're pretty inconsistent about usage of this method, and there's a big mix of using it to print out values and for storing values in the database, and I'm worried that some enterprising developer will come along one day and decide to change the user-facing representation and then break the db stuff.

So I think you're right that `to_string()` should be deprecated and we can root out all of the places we call it internally. Going to avoid doing that in this PR though.

Reply:
maybe we can still swap the implementation though and have `to_string` call `to_db_string`?

Reply:
yep sounds good -- just updated